### PR TITLE
QA Pipelines fixes

### PR DESCRIPTION
**Pull request by mfuntowicz**
**1. Some newly introduced models such as [bart-large-finetuned-squadv1](https://huggingface.co/valhalla/bart-large-finetuned-squadv1) have more than 2 outputs by default on the QA pipeline which is not supported.** 

- This PR makes it possible to support such outputs and assumes the 2 first elements are the actual `start`, `end `logits.

**2. Minor refactoring of the decoding strategy:**

- Actually mask the padding & question **before** applying the softmax to extract answer
- Use the stabilized version of the `softmax `in log-space

### CODE REVIEW

**Comment by thomwolf** [comment]
Maybe we should fix this upstream. I wanted to keep identical behavior for `squad_convert_examples_to_features` while moving the code to the new tokenizer API but maybe I missed something.

**Comment by mfuntowicz** [comment]
@thomwolf I have removed the commit on the padding part to make sure things continue to work at very short term. 

Also, after looking at the code, I've the feeling it requires quite a bit of refactoring that might live in its own PR.

**Review by LysandreJik — APPROVED** [review]
LGTM!

File: src/transformers/pipelines.py [hunk 1]
```diff
@@ -1211,33 +1211,34 @@ def __call__(self, *args, **kwargs):
...
-                end_ = np.exp(end_) / np.sum(np.exp(end_))
-
                 # Mask padding and question
                 start_, end_ = (
                     start_ * np.abs(np.array(feature.p_mask) - 1),
                     end_ * np.abs(np.array(feature.p_mask) - 1),
                 )
 
+                # Mask CLS
+                start_[0] = end_[0] = 0
```
Comment (bryant1410) [inline src/transformers/pipelines.py:1233]: This should be done after "impossible answer" if, otherwise the value returned for that case is not okay.
